docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
+
+Below is the sequence diagram for how the Parser component of `Eduke8` works with commands to show output to the user.
+
+![Parser Sample Sequence](./images/ParserSampleSequence.png)
+
+The command parsing feature is our program’s way of reading the user’s input into the command line. It makes use of a 
+single method `parseCommand` that identifies what command the user is calling for and then calls the command. There are 
+two parsers in our program that implements a single `Parser` interface. One parser is for choosing menu options and is 
+named `MenuParser`. The other parser is used during quizzes, in order to answer questions or request for hints, and is 
+called `QuizParser`. Given below is an example usage scenario of how the command parsing feature works at each step, 
+when the user types in input to get help in order to see what commands are available to the user.
+
+Step 1. The user launches the program for the first time. The `MenuParser()` will be initialised and awaiting the user’s
+        input to proceed.
+        
+Step 2. The user types in "help" into the command line interface and presses enter. This user input “help” is stored as 
+        a string and is put into the `parseCommand()` method as a parameter, together with the list of topics. This 
+        topic list is not relevant to the help command for now.
+        
+Step 3. The user input string is subjected to the `lang.string.trim()` and `lang.string.split()` functions of a string 
+        in the Java libraries in order to remove redundant spaces around the input, and to discern the number of words 
+        in the input. The `lang.string.split()` function uses a blank space string, “ “, as the delimiter to split the 
+        string into its individual components.
+        
+Step 4. Each subsequent string separated by a space is stored in a string array named `commandArr`. The 0th index of the 
+        `commandArr` array is the first word, the 1st index is the second word, and so on. In this case there is only 
+        one word stored in the array, at the 0th index, which is “help”.
+        
+Step 5. The string at the 0th index is then used in a switch statement, where each case represents the different menu 
+        options available. As such, the contents of the case with reference “help” is run, which is a return statement 
+        containing a new `HelpCommand()`. This leads to the execution of the `help` command.
+        
+#### 2.3.3. Design of QuizQuestionsManager
+
+To start a quiz in E-Duke-8, the user will have to indicate the number of questions that he wants to attempt, as well as the topic to get the questions from. Thereafter, questions will be shown to the user one by one until all them are attempted. 
+
+The class diagram given below explains the high-level design of the Quiz system in E-Duke-8. Given below it is a quick overview of each component.
+
+![QuizQuestionsManager_Class_Diagram](./images/QuizQuestionsManager.png)
```

Comment:
Similar to previous comments, could you consider enlarging this diagram? The text is slightly hard to read.

Reply:
![image](https://user-images.githubusercontent.com/47075579/97395598-4445b900-1920-11eb-9054-1139918658f5.png)
Also, for this section of the diagram, the directions of the arrow and the caption do not seem to match. Should the direction of the dotted arrow be swapped, or the caption direction and description altered?

Reply:
Thank you for the feedback! I have enlarged the diagram slightly and also changed the direction of the label arrow!